ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -443,49 +443,49 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
                                         <SlidingPanel isShown={this.showConditions && !!conditions} onClose={() => this.setConditionsStatusVisible(false)}>
                                             {conditions && <ApplicationConditions conditions={conditions} />}
                                         </SlidingPanel>
-                                        <SlidingPanel isShown={!!this.state.metadata} isMiddle={true} onClose={() => this.setState({metadata: null})}>
-                                            {this.state.metadata && (
-                                                <div className='white-box' style={{marginTop: '1.5em'}}>
-                                                    <div className='white-box__details'>
-                                                        <div className='row white-box__details-row'>
-                                                            <div className='columns small-3'>SHA:</div>
-                                                            <div className='columns small-9'>
-                                                                <Revision repoUrl={application.spec.source.repoURL} revision={this.state.metadata.revision} />
+                                        <SlidingPanel isShown={!!this.state.revision} isMiddle={true} onClose={() => this.setState({revision: null})}>
+                                            <DataLoader load={() => services.applications.revisionMetadata(application.metadata.name, this.state.revision)}>
```

Comment:
SlidingPanel content is rendered even when it is not shown. We need to add `this.state.revision && ..`. E.g.:

```
{this.state.revision && (
  <DataLoader load={() => services.applications.revisionMetadata(application.metadata.name, this.state.revision)}>
```